tensorflow/compiler/mlir/tensorflow/tests/lower_tf.mlir

```diff
@@ -248,12 +248,12 @@ func @sqrt_grad_unranked(%arg0: tensor<*xcomplex<f32>>, %arg1: tensor<*xcomplex<
 // %input has 1 batch dimension then 2 block dimensions then 1 remainder
 // dimension.
 // CHECK-LABEL: fourdim_space_to_batch_nd
-func @fourdim_space_to_batch_nd(%input: tensor<3x5x7x10xf32>, %block_shape: tensor<2xi64>, %paddings: tensor<2x2xi64>) -> tensor<?x?x?x10xf32> {
+func @fourdim_space_to_batch_nd(%input: tensor<3x5x7x10xf16>, %block_shape: tensor<2xi64>, %paddings: tensor<2x2xi64>) -> tensor<?x?x?x10xf16> {
```

Comment:
Instead of using an existing test for the new scenario, let's add a new test to keep each test focused on a single scenario and hence simple.

We don't need all the CHECK statements for those and just need to make sure that the test covers the fix. For example, by having CHECK just with the Pad op.

Reply:
Thanks for the suggestion. I did try creating a separate simple test case for this, but it looks really specific and only to catch this resolved bug :) 

Shall we NOT create a new test case, mainly for the following reasons? 

1. There is a similar six dim test case with f32 anyway in the same test file, so an f16 test case is anyway needed, I think.
2. Verifier methods will catch this bug for any input for f16, so if we create a new test case, it will be trivial probably without even a CHECK line.

Reply:
My suggestion was to follow the general testing practice of having each unit test test only one thing. If you notice in this file, the general pattern is that there is first test with most simple and common inputs and then tests for different specific scenarios to improve coverage. It should be possible to create a single test complex test providing full coverage but that is not easy to understand and sometimes fix.

1. That six dim test case is testing specific scenario of higher ranked inputs.
2. If the verifier is catching that, that's good. We can simply have CHECK-LABEL for that.

Reply:
Makes sense. Made the suggested edit.